protocol/messenger_communities.go

```diff
@@ -2516,9 +2570,79 @@ func (m *Messenger) FetchCommunity(request *FetchCommunityRequest) (*communities
 		}
 	}
 
+	if request.Shard == nil {
+		err := m.requestCommunityShardInfoFromMailserver(communityID)
+		if err != nil {
+			return nil, err
+		}
+
+		if !request.WaitForResponse {
+			return nil, nil
+		}
+
+		return m.waitForCommunityResponse(communityID)
+	}
+
 	return m.requestCommunityInfoFromMailserver(communityID, request.Shard, request.WaitForResponse)
 }
 
+// requestCommunityShardInfoFromMailserver requests public shard info by a community ID from the mailserver
+// Note: this is a blocking operation
+func (m *Messenger) requestCommunityShardInfoFromMailserver(communityID string) error {
+
+	m.logger.Info("requesting shard info", zap.String("communityID", communityID))
+
+	m.requestedShardsLock.Lock()
+	defer m.requestedShardsLock.Unlock()
+
+	shardInfoCommunityID := transport.PublicCommunityShardInfoTopic(communityID)
+
+	if _, ok := m.requestedShards[communityID]; ok {
+		return nil
+	}
+
+	//If filter wasn't installed we create it and remember for deinstalling after
+	//response received
+	filter := m.transport.FilterByChatID(shardInfoCommunityID)
+	if filter == nil {
+		filters, err := m.transport.InitPublicFilters([]transport.FiltersToInitialize{{
+			ChatID:      shardInfoCommunityID,
+			PubsubTopic: transport.DefaultShardPubsubTopic(),
+		}})
+		if err != nil {
+			return fmt.Errorf("Can't install filter for community: %v", err)
+		}
+		if len(filters) != 1 {
+			return fmt.Errorf("Unexpected amount of filters created")
+		}
+		filter = filters[0]
+		m.requestedCommunities[communityID] = filter
+	} else {
+		//we don't remember filter id associated with community because it was already installed
+		m.requestedCommunities[communityID] = nil
+	}
+
+	defer m.forgetShardsRequest(communityID)
```

Comment:
tested with commenting this line - didn't help. I still don't get the msg in handle